### PR TITLE
CIV-7140 mediation cya

### PR DESCRIPTION
**Pull request by TatsyMaslova**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-7140

### Change description ###

when mediation is selected by the claimant in lr vs lr set state to in mediation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CaseDataLiP.java [hunk 1]
```diff
@@ -17,4 +18,10 @@ public class CaseDataLiP {
     private TranslatedDocument translatedDocument;
     @JsonProperty("applicant1ClaimMediationSpecRequiredLip")
     private ClaimantMediationLip applicant1ClaimMediationSpecRequiredLip;
+
+    @JsonIgnore
+    public boolean hasClaimantAgreedToFreeMediaiton() {
```
Comment (PaulH-HMCTS) [inline src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CaseDataLiP.java:23]:
```suggestion
    public boolean hasClaimantAgreedToFreeMediation() {
```

File: src/test/java/uk/gov/hmcts/reform/civil/enums/MultiPartyScenarioTest.java [hunk 2]
```diff
@@ -77,23 +87,82 @@ void shouldReturnOneVTwoOneRepWhenTwoRespondentsRepresentedBySameReps() {
...
+    void shouldReturnFalseWhenNotMultiPartyScenario() {
+        CaseData caseData = get1V1CaseData();
+        Assertions.assertFalse(isMultiPartyScenario(caseData));
+
+    }
+
+    @Test
+    void shouldReturnTrueWhenOneToOne() {
+        CaseData caseData = get1V1CaseData();
+        assertThat(isOneVOne(caseData)).isTrue();
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/civil/enums/MultiPartyScenarioTest.java:105]: Maybe just assertTrue/assertFalse on these?

File: src/test/java/uk/gov/hmcts/reform/civil/model/CaseDataTest.java [hunk 3]
```diff
@@ -21,4 +29,97 @@ public void applicant1Proceed_when2v1() {
...
+        //Given
+        CaseData caseData = CaseData.builder().respondent1ClaimResponseTypeForSpec(FULL_DEFENCE).build();
+        //When
+        boolean result = caseData.isRespondentResponseFullDefence();
+        //Then
+        assertThat(result).isTrue();
+    }
+
+    @Test
+    void givenNotOneVTwoTowLegalRepCaseResponsePartAdmit_whenIsRespondentResponseFullDefence_thenFalse() {
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/civil/model/CaseDataTest.java:86]:
```suggestion
    void givenNotOneVTwoTwoLegalRepCaseResponsePartAdmit_whenIsRespondentResponseFullDefence_thenFalse() {
```

File: src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java [hunk 4]
```diff
@@ -480,13 +483,15 @@ public StateFlow build(FlowState.Main initialState) {
                 .transitionTo(CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE).onlyIf(claimDismissedByCamunda)
             .state(CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE)
             .state(FULL_ADMISSION)
```
Comment (neeta-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:485]: https://moj.enterprise.slack.com/files/U042GKV74KZ/F04T3P5Q95X/lr_admissions_flows_v2.1.pdf?origin_team=T1L0WSW9F&origin_channel=D0477PX54GY  as per this flow this change is not required. Please confirm.

File: src/test/java/uk/gov/hmcts/reform/civil/enums/MultiPartyScenarioTest.java (hunk 2 above)
Comment (TatsyMaslova) [reply src/test/java/uk/gov/hmcts/reform/civil/enums/MultiPartyScenarioTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/civil/model/CaseDataTest.java (hunk 3 above)
Comment (TatsyMaslova) [reply src/test/java/uk/gov/hmcts/reform/civil/model/CaseDataTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java (hunk 4 above)
Comment (TatsyMaslova) [reply src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:485]: the flow does not mention full defence. and I am not sure whether during full defence we will or not have mediation. however this story only covers part admit states paid. so i will remove this and when we cover full defence then someone will have to add it if we have mediation. At the moment with full defence we do have mediation but with different fields which is very confusing

Comment (TatsyMaslova) [reply src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:485]: oh sorry Neeta, just had a look and we do have/or should have  mediation screen for full defence, so this flow applies. 
image

**Review by PaulH-HMCTS — DISMISSED** [review]

**Review by neeta-hmcts — DISMISSED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Comment by TatsyMaslova** [comment]
something going wrong with ccd definition import after a few builds so closing this one and reopening a new pr

**Pull request closed without merging**